File: test-connection.js
/**
 * Simple test to check if server is reachable
 */
import axios from 'axios';

const TEST_URL = 'http://localhost:3010/';

async function testConnection() {
  try {
    console.log(`Testing connection to ${TEST_URL}...`);
    const response = await axios.get(TEST_URL);
    console.log('Connection successful!');
    console.log('Response:', response.data);
    return true;
  } catch (error) {
    console.error('Connection failed:', error.message);
    if (error.code) console.error('Error code:', error.code);
    return false;
  }
}

testConnection(); 

File: test_connection.py
#!/usr/bin/env python3

import psycopg2
from datetime import datetime

def log(message):
    timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    print(f"[{timestamp}] {message}")

try:
    log("Testing AWS RDS connection...")
    
    conn = psycopg2.connect(
        host='edsteward-db.cwv4g6g0yzmg.us-east-1.rds.amazonaws.com',
        port=5432,
        database='postgres',
        user='postgres',
        password='EdSteward2024!Secure'
    )
    
    cursor = conn.cursor()
    cursor.execute('SELECT version();')
    version = cursor.fetchone()[0]
    log(f"✅ Connection successful!")
    log(f"PostgreSQL version: {version}")
    
    # Check existing tables
    cursor.execute("SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = 'public';")
    table_count = cursor.fetchone()[0]
    log(f"Current tables in public schema: {table_count}")
    
    cursor.close()
    conn.close()
    
    log("🎉 AWS RDS is ready for database restoration!")
    
except Exception as e:
    log(f"❌ Connection failed: {e}")
    print("\nIf connection failed, please check:")
    print("1. RDS instance is publicly accessible")
    print("2. Security group allows inbound traffic on port 5432")
    print("3. RDS instance is in a public subnet")